src/QsCompiler/Core/Transformations/CommonTransformationNodes.fs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Quantum.QsCompiler.Transformations.Core
+
+open Microsoft.Quantum.QsCompiler.DataTypes
+open Microsoft.Quantum.QsCompiler.SyntaxTree
+
+
+/// <summary>
+/// Contains virtual methods that are invoked
+/// by several different kinds of nodes in the syntax tree.
+/// </summary>
+type CommonTransformationNodes internal () =
```

Comment:
Since this class is only an implementation detail, I don't think it's right to make it a public type (even if the constructor is internal). Instead, I would recommend making it an internal interface. That makes it possible for the transformations to treat `SyntaxTreeTransformation` as implementing `CommonTransformationNodes` but outside code wouldn't know that it existed.

```suggestion
type internal CommonTransformationNodes =
```

Reply:
Agreed in principle. I may be missing something here in terms of F# features, but I don't see how the `inherit ...` would work with this - the edit as suggested doesn't seem to define or allow to call a constructor as part of the SyntaxTreeTransformation class, and `type internal CommonTransformationNodes() = ...` results in an error that the type of the base class is less accessible that the SyntaxTreeTransformation that inherits from it.